sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -3066,34 +3066,38 @@ SIREPO.app.directive('sbatchLoginModal', function() {
               '</div>',
             '</div>',
         ].join(''),
-        controller: function(requestSender, $scope) {
+        controller: function(requestSender, $scope, $rootScope, authState) {
             $scope.otp = '';
             $scope.password = '';
             $scope.username = '';
             var awaitingSendResponse = false;
             var el = $('#sbatch-login-modal');
-            var onHidden = null;
             var errorResponse = null;
+            var onHidden = null;
 
             el.on('hidden.bs.modal', function() {
-                $scope.otp = '';
-                $scope.password = '';
-                $scope.username = '';
-                $scope.sbatchLoginModalForm.$setPristine();
                 var r = {'state': 'error', 'error': 'Please try again.'};
                 if (errorResponse) {
                     r = {'state': 'error', 'error': errorResponse};
                 }
-                errorResponse = null;
+                if (authState.sbatchLoginSuccess) {
+                    $rootScope.$broadcast('sbatchLoginSuccess');
+                }
                 onHidden(r);
                 onHidden = null;
+                errorResponse = null;
+                $scope.otp = '';
+                $scope.password = '';
+                $scope.username = '';
+                $scope.sbatchLoginModalForm.$setPristine();
                 $scope.$apply();
             });
 
             function handleResponse(data) {
-                if (data.hasOwnProperty('state') && data.state == 'error') {
+                if (data.state == 'error') {
```

Comment:
why this change?

Reply:
Checking hasOwnProperty is unnecessary. If data doesn't have state then `data.state === undefined` and `undefined !=== 'error'` so it has the same effect.